### PR TITLE
Added a small shell, needs more apps

### PR DESCRIPTION
**Pull request by bararchy**

### CODE REVIEW

File: kernel/src/lib/nuumshell/shell.cr [hunk 1]
```diff
@@ -0,0 +1,66 @@
+require "./apps/*"
+
+class NuumShell
+  def initialize
+    @user = "root"
+    @pass = "root"
```
Comment (SplittyDev) [inline kernel/src/lib/nuumshell/shell.cr:6]: Is a password needed? At the moment we don't have a filesystem, and a password is essentially useless. :P

File: kernel/src/lib/nuumshell/shell.cr [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @login = false
+    @max_login = 3
```
Comment (SplittyDev) [inline kernel/src/lib/nuumshell/shell.cr:8]: What happens if the max login threshold is reached? Does the user get locked out from the kernel? Seems kinda suboptimal.

File: kernel/src/lib/nuumshell/shell.cr [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  def run
+    banner
+    while !@login
+      login
+    end
+    loop do
+      print "\n#{@user}@nuumsh#: "
+      command = Keyboard.gets.chomp
+      if command.starts_with?("echo")
```
Comment (SplittyDev) [inline kernel/src/lib/nuumshell/shell.cr]: Maybe split the command using `String#split` and only check the first item?
For performance reasons, splitting with a limit of 2 would probably be most efficient.

Maybe something like that:
```crystal
input = Keyboard.gets.chomp
command_parts = input.split 2
if command_parts.size != 0
  command = command_parts[0]
  # ...
end
```

Just an idea though, you don't need to change that if you don't like it.

File: kernel/src/lib/nuumshell/shell.cr [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      print "Login: "
+      user = Keyboard.gets.chomp
+      print "Password: "
+      pass = Keyboard.gets(:silent).chomp
+      if pass == @pass && user == @user
+        @login = true
+        break
+      end
+      puts "User or password are incorrect"
+      try = +1
```
Comment (SplittyDev) [inline kernel/src/lib/nuumshell/shell.cr]: This is most likely a bug. You are not incrementing `try`, you are only setting it to positive `1`.

File: kernel/src/lib/nuumshell/shell.cr (hunk 1 above)
Comment (bararchy) [reply kernel/src/lib/nuumshell/shell.cr:6]: It's more of a cool nice to have feature :) I can remove if you want

File: kernel/src/lib/nuumshell/shell.cr (hunk 2 above)
Comment (bararchy) [reply kernel/src/lib/nuumshell/shell.cr:8]: Nope, just the login procedure is restarting, and when we have more stuff (like a FS) a log can be created for this event

**Comment by bararchy** [comment]
@SplittyDev Can you take another look ? should look better now

**Review by SplittyDev — APPROVED** [review]

**Pull request merged**